### PR TITLE
arm-frc-linux-gnueabi-gcc: init at 4.9.4

### PR DESCRIPTION
**Pull request by colescott**
###### Motivation for this change
This adds support for FIRST Robotics Competition compiling to nix by adding a cross compiler and all necessary packages. Normally the compiler will take a lot of work to build in the userspace, but with this package it is all automated and compatible with nix. 

This package is based on the arch version of the package: https://aur.archlinux.org/packages/arm-frc-linux-gnueabi-gcc/

DOES NOT SUPPORT MACOS
Support for macOS can be found here: http://first.wpi.edu/FRC/roborio/toolchains/

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] ~macOS~ 
   - [x] Linux
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/compilers/arm-frc-linux-gnueabi-gcc/default.nix [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  src = fetchurl {
+    url = "ftp://gcc.gnu.org/pub/gcc/releases/gcc-${version}/gcc-${version}.tar.bz2";
+    sha256 = "6c11d292cd01b294f9f84c9a59c230d80e9e4a47e5c6355f046bb36d4f358092";
+  };
+
+  patches =
+  [
+    ./minorSOname.patch
+    ./no-nested-deprecated-warnings.patch
+    ./format-error-fix.patch
```
Comment (Mic92) [inline pkgs/development/compilers/arm-frc-linux-gnueabi-gcc/default.nix]: An alternative would have been to just disable format warnings `hardeningDisable = [ "format" ];`

**Comment by colescott** [comment]
Travis build failed due to 4MB log limit >.<

File: pkgs/development/compilers/arm-frc-linux-gnueabi-gcc/default.nix (hunk 1 above)
Comment (colescott) [inline pkgs/development/compilers/arm-frc-linux-gnueabi-gcc/default.nix]: I am going to update it to include that and remove the patch. No need for all the extra file size on the repo.

Comment (colescott) [inline pkgs/development/compilers/arm-frc-linux-gnueabi-gcc/default.nix]: Fixed.

**Comment by Mic92** [comment]
yes, just keep the log as it is. Travis is only used here for pull requests.

**Comment by colescott** [comment]
Could a macOS person test this please? I don't have access to osx atm

**Comment by colescott** [comment]
macOS support removed because toolchain is available for OSX at http://first.wpi.edu/FRC/roborio/toolchains/

ar and tar were a bit of a pain here.


I think it is ready to be merged

**Comment by colescott** [comment]
@Mic92 It won't pass Travis due to log size but it is ready to be merged.

**Comment by Mic92** [comment]
I rebased the patches here: https://github.com/Mic92/nixpkgs/commits/frc
If everything is still working for you, I would push this to master.

**Comment by matthewbauer** [comment]
Is there a way we can do a compatibility alias for this? Not sure what the "equivalent" would be for arm-frc-linux-gnueabi-gcc

**Comment by Ericson2314** [comment]
@matthewbauer Yes, we can do it like `pkgsi686Linux`. See https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L20-L28 and the use of `nixpkgsFun` to recur. Also see https://github.com/NixOS/nixpkgs/blob/master/pkgs/top-level/all-packages.nix#L5640-L5665 for the one for cross in particular. I was hoping to get rid of `nixpkgsFun` eventually, but I suppose the compat alias will only be temporary.

**Comment by Ericson2314** [comment]
Hehe it's like exactly a year so maybe that competition is about to begin again anew!

**Comment by matthewbauer** [comment]
Lol- I think FIRST starts in January though

**Pull request closed without merging**